src/fiber/fiber.mli

```diff
@@ -322,6 +317,36 @@ module Stream : sig
 end
 with type 'a fiber := 'a t
 
+module Pool : sig
+  (** Pool is used to submit asynchronous tasks without waiting for their
+      completion. *)
+  type t
+
+  type 'a fiber
+
+  (** Create a new pool. *)
+  val create : unit -> t
+
+  (** [running pool] returns whether it's possible to submit tasks to [pool] *)
+  val running : t -> bool fiber
+
+  (** [task pool ~f] submit [f] to be done in [pool]. Errors raised [pool] will
+      not be raised in the current fiber, but inside the [Pool.run] fiber.
+
+      If [running pool] returns [false], this function will raise a
+      [Code_error]. *)
+  val task : t -> f:(unit -> unit fiber) -> unit fiber
+
+  (** [stop pool] stops the pool from receiving new tasks. After this function
+      is called, [task pool ~f] will always raise a [Code_error]. *)
```

Comment:
Thanks for adding the doc. I removed the document of `Code_error` being raised; catching such an exception would be bad practice, so we shouldn't document it.

One thing I realised while reading the doc: when stop is called, no new task will be accepted but already submitted tasks that haven't started will still be fired. Is that the behaviour you want?

Reply:
Yes, that behavior makes sense to me. Already submitted tasks should run.

Otherwise, the callers wouldn't have a way of knowing if his task was discarded or not.

Reply:
That's a good point. We should document this behaviour then, as it might not be obvious.

Reply:
The documentation of `stop` is the last remaining point on this PR then. The rest looks good. Feel free to merge once you have updated the doc (and CI passes, as usual).